src/main/java/frc/robot/Commands/ReturnByGyro.java

```diff
@@ -0,0 +1,54 @@
+package frc.robot.Commands;
+
+import com.spikes2212.command.drivetrains.commands.DriveArcadeWithPID;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import frc.robot.Subsystems.Dispenser;
+import frc.robot.Subsystems.Drivetrain;
+import com.spikes2212.util.Limelight;
+
+/**
+ * This command moves the {@link Drivetrain} till either its gyro angle is zero or the
+ * {@link Limelight} has a valid target.
+ *
+ * @author Ofri Rosenbaum
+ */
+public class ReturnByGyro extends DriveArcadeWithPID {
+
+    private static final double SPEED = 0.7;
+    private static final int LIMELIGHT_PIPELINE = 1;
+
+    /**
+     * The gyro's tolerance.
+     */
+    private static final double TOLERANCE = 2;
+
+    private PIDSettings pidSettings;
+    private FeedForwardSettings feedForwardSettings;
+
+    private Dispenser dispenser;
+
+    private int previousPipeline;
+
+    public ReturnByGyro(Drivetrain drivetrain, PIDSettings pidSettings, FeedForwardSettings feedForwardSettings) {
```

Comment:
get PIDSettings and FFSettings from `drivetrain`

Reply:
Say we want pid for distance and not only angle so this will have to be two different pids, should we put them both in drivetrain?